pkg/chains/formats/sbom/sbom.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sbom
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"k8s.io/client-go/kubernetes"
+)
+
+const maxSBOMBytes = 5 * 1024 * 1024 // 5 Megabytes
```

Comment:
Why 5 Megabytes? Are most SBOMs much smaller than this?

Reply:
I don't have the stats. The reasoning here was if it's over 5MBs, there's something wrong with your SBOM :) 

This should probably be configurable though.

Reply:
@lcarva FWIW, I have a testing cyclonedx sbom that has around 1700 components and the size is 5.2M ;)

Reply:
My goodness.. So 5mb is not _big_ enough. I'll update this PR to make it configurable and set a modest default value to 10MB.

Reply:
Yeah, I was surprised myself when I checked. That sbom was for a Service Registry image. 10MB sounds good. But of course it's hard to say how often users will have sboms bigger than that.

Reply:
Updated the PR to make it configurable via the key `artifacts.sbom.maxbytes`. Set the default to 10 MB. Documented.